tests/test_SklearnGradientBoostingConverters.py

```diff
@@ -6,22 +6,49 @@
 
 import unittest
 import numpy as np
-from sklearn.datasets import make_regression
+from pandas import DataFrame
+from sklearn.datasets import make_regression, make_classification
 from sklearn.ensemble import GradientBoostingClassifier
 from sklearn.ensemble import GradientBoostingRegressor
 from skl2onnx import convert_sklearn
 from skl2onnx.common.data_types import FloatTensorType
 from skl2onnx.common.data_types import onnx_built_with_ml
 from test_utils import dump_binary_classification, dump_multiple_classification
 from test_utils import dump_data_and_model
+from onnxruntime import InferenceSession
 
 
 class TestSklearnGradientBoostingModels(unittest.TestCase):
     @unittest.skipIf(not onnx_built_with_ml(),
                      reason="Requires ONNX-ML extension.")
-    def test_gradient_boosting_classifier(self):
+    def test_gradient_boosting_classifier1Deviance(self):
+        model = GradientBoostingClassifier(n_estimators=1, max_depth=2)
+        X, y = make_classification(10, n_features=4, random_state=42)
+        X = X[:, :2]
+        model.fit(X, y)
+
+        for cl in [1e-6, 0.9, 0.5, 0., 1e-5]:
+            model.init_.class_prior_ = np.array([cl, cl])
+            initial_types = [('input', FloatTensorType((1, X.shape[1])))]
+            model_onnx = convert_sklearn(model, initial_types=initial_types)
+            if "Regressor" in str(model_onnx):
+                raise AssertionError(str(model_onnx))
+            sess = InferenceSession(model_onnx.SerializeToString())
+            res = sess.run(None, {'input': X.astype(np.float32)})
+            pred = model.predict_proba(X)
+            if res[1][0][0] != pred[0, 0]:
```

Comment:
Why are you not using dump_data_and_model() here?

Reply:
See below.